static/js/app.ts

```diff
@@ -1961,7 +1961,14 @@ function programNeedsSaving(adventureName: string) {
   const localStorageCanBeSavedToServer = theUserIsLoggedIn && adventure.save_info === 'local-storage';
   const isUnchangeable = isServerSaveInfo(adventure.save_info) ? adventure.save_info.submitted : false;
 
-  return (programChanged || nameChanged || localStorageCanBeSavedToServer) && !isUnchangeable;
+  // Do not autosave the program if the size is very small compared to the previous
+  // save. This protects against accidental `Ctrl-A, hit a key` and everything is gone. Clicking the
+  // "Run" button will always save regardless of size.
+  const wasSavedBefore = adventure.save_info !== undefined;
+  const suspiciouslySmallFraction = 0.1;
```

Comment:
```suggestion
  const suspiciouslySmallFraction = 0.1;
```

Thanks for this fix, this is good protection against accidental erase. But 10% might be a bit low? I suspect that a drop like 50% already is something dubious. Who would erase half their program (esp in the presence of 😴 to remove lines from execution)